common/djangoapps/third_party_auth/models.py

```diff
@@ -288,7 +306,7 @@ class SAMLProviderConfig(ProviderConfig):
     Configuration Entry for a SAML/Shibboleth provider.
     """
     prefix = 'saml'
-    KEY_FIELDS = ('idp_slug', )
+    KEY_FIELDS = ('site_id', 'idp_slug',)
```

Comment:
Yes, I like this idea much better than adding the site keys. I reverted most of the changes I've made since doing so and will try to take this approach instead.